metadata.c

```diff
@@ -152,16 +152,48 @@ static int pv_devmeta_uname(struct pv_devmeta_read *pv_devmeta_read)
 	if (pv_devmeta_buf_check(pv_devmeta_read))
 		return -1;
 
+	char out[2048] = {0};
 	char err[2048] = {0};
-	tsh_run_output("uname -a", 2, buf, buflen, err, 2048);
+	tsh_run_output("uname -a", 2, out, 2048, err, 2048);
 
-	if (strnlen(buf, buflen) == buflen) {
+  if (strnlen(out, 2048) == 0) {
 		memset(buf, 0, buflen);
 		return -1;
 	}
 
+	// removes trailing new line character
+	out[strlen(out) - 1] = '\0';
+
+	char *kname = &out[0];
+	char *node = strchr(kname, ' ');
+	*node = '\0';
+	++node;
+
+	char *krelease = strchr(node, ' ');
+	*krelease = '\0';
+	++krelease;
+
+	char *kversion = strchr(krelease, ' ');
+	*kversion = '\0';
+	++kversion;
+
+	char *os = strrchr(kversion, ' ');
+	*os = '\0';
+	++os;
+
+	char *arch = strrchr(kversion, ' ');
+	*arch = '\0';
+	++arch;
+
```

Comment:
is splitting the uname -a output in this order always correct?

i would suggest we rather use the posix c function uname() ... see man 2 uname...

Reply:
Fixed in a39869406c384d6e2714349ab06e21bc4afdc311